AsyncDisplayKit/Layout/ASLayout.mm

```diff
@@ -253,3 +229,18 @@ - (NSString *)_recursiveDescriptionForLayout:(ASLayout *)layout level:(NSUIntege
 }
 
 @end
+
+ASLayout *ASCalculateLayout(id<ASLayoutable> layoutable, const ASSizeRange sizeRange, const CGSize parentSize)
+{
+  ASDisplayNodeCAssertNotNil(layoutable, @"Not valid layoutable passed in.");
+  
+  return layoutable ? [layoutable layoutThatFits:sizeRange parentSize:parentSize] : nil;
```

Comment:
@rahul: We cannot return the `layoutable` here so the shorter version would be wrong, as if a `layoutable` is `!= nil` we have to create an `ASLayout` and not an `id<ASLayoutable>` via `layoutThatFits:parentSize`

@nguyenhuy: We could do that. You are right that a `layoutable` should not be nil in this case, but theoretically it can happen ( and it happened from past experience :) . That said, calling a method on nil will return nil anyway so it should be fine and changed.